code/workspaces/infrastructure-api/src/kubernetes/deploymentGenerator.js

```diff
@@ -1,20 +1,19 @@
-import fs from 'fs';
-import yaml from 'js-yaml';
+import { imageConfig, image, defaultImage } from 'common/src/config/images';
 import { DeploymentTemplates, ServiceTemplates, generateManifest, ConfigMapTemplates } from './manifestGenerator';
 import nameGenerator from '../common/nameGenerators';
 import config from '../config/config';
 
-const containerInfo = yaml.safeLoad(fs.readFileSync(config.get('containerInfoPath')));
+const containerInfo = imageConfig();
 
-function createJupyterDeployment({ projectKey, deploymentName, notebookName, type, volumeMount }) {
+function createJupyterDeployment({ projectKey, deploymentName, notebookName, type, volumeMount, version }) {
   const startCmd = type === 'jupyterlab' ? 'lab' : 'notebook';
+  const img = version == null ? defaultImage(type) : image(type, version);
```

Comment:
Is it worth the version checks being a generic check for falsy values? Would this be more robust to empty strings etc. being provided as a version?

```js
  const img = version ? defaultImage(type) : image(type, version);
```